crates/cli-support/src/js/mod.rs

```diff
@@ -322,12 +322,33 @@ impl<'a> Context<'a> {
         // Deno removed support for .wasm imports in https://github.com/denoland/deno/pull/5135
         // the issue for bringing it back is https://github.com/denoland/deno/issues/5609.
         format!(
-            "const file = new URL(import.meta.url).pathname;
-            const wasmFile = file.substring(0, file.lastIndexOf(Deno.build.os === 'windows' ? '\\\\' : '/') + 1) + '{}_bg.wasm';
-            const wasmModule = new WebAssembly.Module(Deno.readFileSync(wasmFile));
-            const wasmInstance = new WebAssembly.Instance(wasmModule, imports);
+            "const url = new URL(import.meta.url);
+
+            let wasmCode = '';
+            switch (url.protocol) {{
+                case 'file:':
+                    let file = new URL(import.meta.url).pathname;
+                    if (Deno.build.os === 'windows') {{
+                        if (file.startsWith('/')) file = file.substr(1);
+                        file = await Deno.realPath(file + '/../{module_name}_bg.wasm');
+                    }} else {{
+                        file = file.substring(0, file.lastIndexOf('/') + 1) + '{module_name}_bg.wasm';
+                    }}
+                    wasmCode = await Deno.readFile(file);
+                    break
+                case 'https:':
+                case 'http:':
+                    const wasm_url = import.meta.url.substring(0, import.meta.url.lastIndexOf('/') + 1) + '{module_name}_bg.wasm';
+                    wasmCode = await (await fetch(wasm_url)).arrayBuffer();
+                    break
+                default:
+                    throw `Unsupported protocol: ${{url.protocol}}`;
```

Comment:
Throwing strings is a bad practice, since they don't have any stacktrace attached. Best to throw a proper `Error` instance:

```suggestion
                    throw new Error(`Unsupported protocol: ${{url.protocol}}`);
```